qt/applications/workbench/workbench/widgets/settings/general/test/test_general_settings.py

```diff
@@ -158,9 +158,9 @@ def test_load_current_setting_values(self, mock_ConfigService, mock_CONF):
 
         # calls().__int__() are the calls to int() on the retrieved value from ConfigService.getString
         mock_CONF.get.assert_has_calls([call(GeneralSettings.PROMPT_SAVE_ON_CLOSE),
-                                        call().__int__(),
+                                        call().__index__(),
```

Comment:
From the tests it looks like there is a difference between Python 3.8 and lower. It must be Python 3.8 that now calls `__index__` and everything older calls `__int__`. I think we'll have to add a check on the Python version. Something like

```python
if sys.version_info < (3, 8):
     # do __int__ check
else:
    # do __index__ check